libdirections-offboard/src/main/java/com/mapbox/navigation/route/offboard/MapboxOffboardRouter.kt

```diff
@@ -20,41 +15,45 @@ class MapboxOffboardRouter(
 ) : Router {
 
     private var navigationRoute: NavigationRoute? = null
+    // For testing only
+    private var isNeedToBuildNavigationRoute = true
+    // For testing only
+    private lateinit var routeRetrieveCallback: RouteRetrieveCallback
```

Comment:
Forcing behavior for tests only often means that the design of the class is incorrect.

Reply:
@LukasPaczos Yeah, I'm totally agree with you. But I thought we agreed with this class architecture and interfaces, so I shouldn't change their signature. And I didn't invent how to avoid this stuff while you are not provide it in comments below) So thanks a lot for your ideas, I'll review them and fix your comments 👍